backend/src/agents/answer_agent.py

```diff
@@ -0,0 +1,22 @@
+from datetime import datetime
+from typing import Optional
+from src.utils import call_model
+from src.prompts import PromptEngine
+from src.agents import Agent, agent
+
+engine = PromptEngine()
+intent_format = engine.load_prompt("intent-format")
+
+datetime = datetime.now()
```

Comment:
This may be in the slightly wrong position. I wonder whether this would be computed once when this agent is imported to the director. Would we want it to be the exact datetime when the answer agent is invoked, and if so then I feel like this should be inside the invoke? (I may be barking up the wrong tree hear!)